Assets/Scripts/GameManager/GameManager.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+/// <summary>
+/// handles Global game preferences and state
+///
+/// uses Singleton pattern, with thread lock and lazy instantiation
+/// </summary>
+public sealed class GameManager
+{
+    private const int LivesPerLevel = 3; // Maximum lives per level
+    private const int RewardPointForCollectible = 10;
+
+    private int _livesRemaining = LivesPerLevel;
+    private int _currentLevel = 1;
+    private int _currentLevelScore = 0; // current score of user per level
+
+    private readonly List<IScoreChangedListener> _scoreChangedListeners;
+    private readonly List<ILivesCountChangeListener> _livesCountChangeListeners;
+
+
+    /// <summary>
+    /// no instances allowed 
+    /// </summary>
+    private GameManager()
+    {
+        _scoreChangedListeners = new List<IScoreChangedListener>();
+        _livesCountChangeListeners = new List<ILivesCountChangeListener>();
+    }
+
+    private static readonly Lazy<GameManager> LazyGameManager = new Lazy<GameManager>(() => new GameManager());
+    public static GameManager Instance => LazyGameManager.Value;
+
+
+    public void AddLivesCountChangeListener(ILivesCountChangeListener listener)
+    {
+        _livesCountChangeListeners.Add(listener);
+        listener.OnLivesCountChanged(_livesRemaining);
+    }
+
+    public void RemoveLivesCountChangeListener(ILivesCountChangeListener listener)
+    {
+        _livesCountChangeListeners.Remove(listener);
+    }
+
+
+    public void AddScoreChangeListener(IScoreChangedListener listener)
+    {
+        _scoreChangedListeners.Add(listener);
+        listener.OnScoreChanged(_currentLevelScore);
+    }
+
+    public void RemoveScoreChangeListener(IScoreChangedListener listener)
+    {
+        _scoreChangedListeners.Remove(listener);
+    }
+
+
+    public void OnLevelCompleted()
+    {
+        _currentLevelScore += 50;
```

Comment:
Do not use constant values like 50 . Use it in the starting of the class with private serialize field